sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/NoSuchItemException.scala

```diff
@@ -40,10 +41,12 @@ class NoSuchPartitionException(
 class NoSuchPermanentFunctionException(db: String, func: String)
   extends AnalysisException(s"Function '$func' not found in database '$db'")
 
-class NoSuchFunctionException(db: String, func: String)
+class NoSuchFunctionException(db: String, func: String, cause: Option[Throwable] = None)
   extends AnalysisException(
-    s"Undefined function: '$func'. This function is neither a registered temporary function nor " +
-    s"a permanent function registered in the database '$db'.")
+     s"Undefined function: '$func'. This function is neither a registered temporary function nor " +
+      s"a permanent function registered in the database '$db'." +
+      s"${cause.map(th => s"Exception thrown during look up:" +
```

Comment:
This is too complex for string interpolation. But why not just set the exception's cause?

Reply:
I agree. Original implementation is just set exception's cause.

Reply:
I mean, why not set the cause on `AnalysisException`? the cause may be null if the Option is None, that's fine.

Reply:
Update @srowen Thanks!